jOOQ/src/main/java/org/jooq/tools/jdbc/MockStatement.java

```diff
@@ -188,7 +188,8 @@ private boolean execute0(
         );
 
         result = data.execute(context);
-        return result != null && result.length > 0 && result[resultIndex].data != null;
+
+        return localSql.startsWith("select");
```

Comment:
There are many ways in which this will not be robust enough, among which
- there could be a parenthesis (or several) leading the  `select` keyword
- the keyword might not be in lower case
- there might be a comment prepended to the keyword
- there might be a `with` clause prepended to `select`